src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -219,12 +219,19 @@ public static bool IsAsyncDisposeImplementation([NotNullWhen(returnValue: true)]
                 method.IsImplementationOfInterfaceMethod(null, iAsyncDisposable, "DisposeAsync");
         }
 
+        /// <summary>
+        /// Checks the name of the method of an implicit interface implementation (so, as is) or an explicit interface implementation (which also contains the interface as a prefix). 
+        /// </summary>
+        private static bool IsImplicitOrExplicitInterfaceImplementationName(this IMethodSymbol method, string name, string interfacePrefix)
+            => (method.Name == name && method.MethodKind is MethodKind.Ordinary) ||
+            (method.Name == $"{interfacePrefix}.{name}" && method.MethodKind is MethodKind.ExplicitInterfaceImplementation);
```

Comment:
I think the correct way to check explicit interface implementation is to check the name for the explicit implementation method, not the method itself. `IsImplicitOrExplicitInterfaceImplementationName` method should take an `INamedTypeSymbol disposableType` parameter (the argument could be the type symbol for `System.IDisposable` or type symbol for `System.IAsyncDisposable) instead of `string interfacePrefix` and the check should be as below:

```csharp
private static bool IsImplicitOrExplicitInterfaceImplementationName(this IMethodSymbol method, string name, INamedTypeSymbol disposableType)
            => (method.Name == name && method.MethodKind is MethodKind.Ordinary) ||
            (method.MethodKind is MethodKind.ExplicitInterfaceImplementation && method.ExplicitInterfaceImplementations.Any(i => i.Name == name && SymbolEqualityComparer.Default.Equals(i.ContainingType, disposableType))
```

Reply:
I would also convert the above into a block body instead of expression body for clarity:

```csharp
private static bool IsImplicitOrExplicitInterfaceImplementationName(this IMethodSymbol method, string name, INamedTypeSymbol disposableType)
{
    switch (method.Kind)
    {
        case MethodKind.Ordinary:
            return method.Name == name;

        case MethodKind.ExplicitInterfaceImplementation:
            return method.ExplicitInterfaceImplementations.Any(i => i.Name == name && SymbolEqualityComparer.Default.Equals(i.ContainingType, disposableType);

        default:
            return false;
    }
}
```

Reply:
Actually, scratch all the above comments. I looked at primary entrypoint method to get Dispose method kind and it seems it already handles the explicit implementation case: https://github.com/dotnet/roslyn-analyzers/blob/c2bc164fe991d87662dc5b6d3aa94bf276392241/src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs#L313-L359

So, if the tests you added fail, then something needs to be fixed in the above method itself?